python-backend/app/api/mines/reports/resources/mine_report_comment.py

```diff
@@ -0,0 +1,132 @@
+    @api.doc(description='creates a new comment for the report submission')
+    @api.marshal_with(MINE_REPORT_COMMENT_MODEL, code=201)
+    @requires_role_edit_report
+    def post(self, mine_guid, mine_report_guid):
+
+        mine_report_submission = MineReportSubmission.find_latest_by_mine_report_guid(
+            mine_report_guid)
+
+        if not mine_report_submission:
+            raise NotFound('Mine report submission not found')
+            # TODO: Do we want to create a submission if it doesn't exist?
+
+        data = self.parser.parse_args()
+
+        if not data['report_comment']:
+            raise BadRequest('Empty comment')
+
+        mine_report_comment_guid = uuid.uuid4()
+
+        mine_report_comment = MineReportComment.create(
+            mine_report_submission,
+            mine_report_comment_guid=mine_report_comment_guid,
+            report_comment=data['report_comment'],
+            comment_visibility_ind=data['comment_visibility_ind'],
+        )
+
+        current_app.logger.info(f'Creating comment {mine_report_comment}')
+
+        try:
+            mine_report_comment.save()
+        except Exception as e:
+            raise InternalServerError(f'Error when saving: {e}')
+
+        return mine_report_comment, 201
+
+
+class MineReportCommentResource(Resource, UserMixin):
+    parser = CustomReqparser()
+    parser.add_argument('report_comment', type=str, location='json')
+    parser.add_argument('comment_visibility_ind', type=inputs.boolean, location='json')
+
+    @api.expect(MINE_REPORT_COMMENT_MODEL)
+    @api.doc(description='update a comment')
+    @api.marshal_with(MINE_REPORT_COMMENT_MODEL, code=201)
+    @requires_role_edit_report
+    def put(self, mine_guid, mine_report_guid, mine_report_comment_guid=None):
+
+        data = self.parser.parse_args()
+        comment = MineReportComment.find_by_guid(mine_report_comment_guid)
+        if not comment:
+            raise NotFound('Mine Report Comment not found.')
+
+        current_app.logger.info(f'Updating {comment} with {data}')
+        for key, value in data.items():
+            setattr(comment, key, value)
+
+        try:
+            comment.save()
+        except Exception as e:
+            raise InternalServerError(f'Error when saving: {e}')
+
+        return comment, 201
+
+    @api.doc(
+        description='Delete a mine report comment by guid', params={'mine_report_comment_guid': 'guid of the comment to delete.'})
+    @requires_role_mine_admin
+    def delete(self, mine_guid, mine_report_guid, mine_report_comment_guid):
+        if mine_report_comment_guid is None:
+            return self.create_error_payload(404, 'Must provide a mine report comment guid.'), 404
+        try:
+            comment = MineReportComment.find_by_guid(mine_report_comment_guid)
+        except DBAPIError:
+            return self.create_error_payload(422, 'Invalid Mine Report Comment guid'), 422
+        if comment is None:
```

Comment:
Does this work? It looks like it should give a `comment was referenced before it was defined` error since comment is defined in a `try` block